src/clearwater_modules/tsm/processes.py

```diff
@@ -412,39 +400,29 @@ def mf_density_air_sat(water_temp_k: float, esat_mb: float, pressure_mb: float)
     return 0.348 * (pressure_mb / water_temp_k) * (1.0 + mixing_ratio_sat) / (1.0 + 1.61 * mixing_ratio_sat)
 
 
-@numba.njit
-def mf_cp_water(water_temp_c: float) -> float:
+def mf_cp_water(water_temp_c: xr.DataArray) -> xr.DataArray:
     """
     Compute the specific heat of water (J/kg/K) as a function of water temperature (Celsius).
     This is used in computing the source/sink term.
     """
-    # TODO: make this work as a ufunc
-    return water_temp_c * 2
-    if water_temp_c <= 0.0:
-        return 4218.0
-    elif water_temp_c <= 5.0:
-        return 4202.0
-    elif water_temp_c <= 10.0:
-        return 4192.0
-    elif water_temp_c <= 15.0:
-        return 4186.0
-    elif water_temp_c <= 20.0:
-        return 4182.0
-    elif water_temp_c <= 25.0:
-        return 4180.0
-    else:
-        return 4178.0
+    return xr.where(water_temp_c <= 0.0, 4218.0,
+        xr.where(water_temp_c <= 5.0, 4202.0,
+        xr.where(water_temp_c <= 10.0, 4192.0,
+        xr.where(water_temp_c <= 15.0, 4186.0,
+        xr.where(water_temp_c <= 20.0, 4182.0,
+        xr.where(water_temp_c <= 25.0, 4180.0, 4178.0
+    ))))))
```

Comment:
This function had been implemented as as a lookup table, but this is rather inaccurate relative to a continuous function, such as a polynomial or even a linear interpolation. We can implement this for now but should create an issue to replace this function.

Reply:
(((4.65e-6*water_temp_c - 0.001)*water_temp_c + 0.085858)*water_temp_c - 3.1331)*water_temp_c + 4219.793

valid for temperature 0 through 80. Do we need to go higher?

Reply:
@PeterKlaver, thanks! Did that equation come from the [Zhong & Johnson 2016a](https://hdl.handle.net/11681/10112) report?

Reply:
It did not. There is no explicit discussion of how Cp is calculated in section 2 of that document, although it includes this statement:  "Detailed discussion of equations and mechanisms can be found from Water Resources Engineers Inc. (1967), Brown and Barnwell (1987), and Deas and Lowney (2000). Much of this section was adopted from HEC (2010a) to match the kinetic implementations in original water temperature model."

I'm assuming the lookup table came directly from Zhong's NSM. 

The equation I provided was just a regression I did from tabulated values on the Engineering Toolbox website, as a proof of concept. I can easily repeat this with values from a thermodynamics textbook, which we could then reference directly.

Reply:
So should I just implement the equation @PeterKlaver commented @aufdenkampe ? I feel that maybe 80 degrees C is too restrictive?

Reply:
I did some looking into the practical working range for these kind of heat balance formulations, in models like CE-QUAL-W2 and QUAL2k, but I couldn't find anything explicit. I should think, then, that common sense can be a guide, and that anything much above 40 deg C is an unrealistic condition for natural waters. So if we provide a good number for Cp up to 80 we are exceeding expectations (especially considering the original table stopped at 25).

Reply:
@aufdenkampe I used the column for isobaric (constant pressure) with units of kJ/kg-deg K. I multiplied those by 1000 to get the same units as the original table. The equation assumes that the water temperature is in deg C.

Reply:
Polynomial was implemented w/ commit 59af362e005a63eef8b61fa10d0ac1f39c951ebe